src/contracts/pods/EigenPod.sol

```diff
@@ -428,6 +451,96 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+
+
+        eigenPodManager.requestProofViaSuccinctGateway(
+            RANGE_SPLITTER_FUNCTION_ID, 
+            _timestampToSlot(timestampProvenUntil),
+            _timestampToSlot(endTimestamp),
+            address(this),
+            callBackData,
+            callbackGasLimit
+        );
+    }
+
+    /// @notice The callback function for the ZK proof fulfiller.
+    function handleCallback(uint256 nonce, uint64 oracleTimestamp, uint64 startSlot, uint64 endSlot) external onlySuccinctGateway() {
+        PartialWithdrawalProofRequest memory request = _partialWithdrawalProofRequests[nonce];
+
+        require(nonce == lastRequestNonceProven + 1, "EigenPod.handleCallback: must callback in order");
+        require(startSlot < endSlot, "EigenPod.handleCallback: invalid start and end slot values");
+        require(_slotToTimestamp(endSlot) <= request.endTimestamp, "EigenPod.handleCallback: endSlot must be less than the request's endTimestamp");
+        require(_slotToTimestamp(startSlot) >= timestampProvenUntil, "EigenPod.handleCallback: startSlot must be greater than or equal to the timestampProvenUntil");
+        require(request.status == REQUEST_STATUS.PENDING, "EigenPod.handleCallback: request nonce is either cancelled or fulfilled");
+
+        bytes32 beaconBlockRoot = eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp);
+        bytes memory output = eigenPodManager.confirmProofVerification(WITHDRAWAL_FUNCTION_ID, abi.encodePacked(beaconBlockRoot, address(this), startSlot, endSlot));
+
+        uint256 partialWithdrawalSumWei = abi.decode(output, (uint256));
+        //record the timestamp until which all withdrawals have been proven
+        timestampProvenUntil = _slotToTimestamp(endSlot);
+        emit PartialWithdrawalProven(nonce, partialWithdrawalSumWei);
+
+        //subtract out any partial withdrawals proven via merkle proofs in the interim
+        uint256 amountToSendWei;
+
+        if(sumOfPartialWithdrawalsClaimedWei > partialWithdrawalSumWei){
+            sumOfPartialWithdrawalsClaimedWei -= partialWithdrawalSumWei;
+        } else {
+            amountToSendWei = partialWithdrawalSumWei - sumOfPartialWithdrawalsClaimedWei;
+            /**
+            * Zero out the running total of partial withdrawals claimed. Any merkle proofs for partial withdrawals proven
+            * after this this request for a proof is made will be added to sumOfPartialWithdrawalsClaimedWei.  When the
+            * zk proof is fulfilled, we will subtract sumOfPartialWithdrawalsClaimedWei from the amount proven by the oracle
+            */
+            sumOfPartialWithdrawalsClaimedWei = 0;
+        }
+
+        //mark the partial withdrawal proof as being fulfilled
+        if(endSlot == request.endTimestamp){
+            _partialWithdrawalProofRequests[nonce].status = REQUEST_STATUS.FULFILLED;
+            lastRequestNonceProven = nonce;
+        }
```

Comment:
ahh, so this is the current approach you have to batching?
I see what you were talking about now -- if batched callbacks aren't in order, then it's a major issue as partial withdrawals will get stuck!

Reply:
exactly!  If we had separate nonces for each split, we could enforce they are made in order in the handleCallback function. This would be way better imo